[stable/Jenkins] Added jenkins location configuration

rbutcher opened the pull request:
<!--
Thank you for contributing to kubernetes/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/kubernetes/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

**What this PR does / why we need it**:
Added jenkins.model.JenkinsLocationConfiguration.xml to config map to support setting JENKINS_URL to avoid warnings on first start and manual configuration.

**Which issue this PR fixes**:
fixes #6513

rbutcher committed:
Added jenkins location configuration

- Added jenkins.model.JenkinsLocationConfiguration.xml to add the
ability to set the jenkinsUrl
- Added line to copy jenkins.model.JenkinsLocationConfiguration.xml into
JENKINS_HOME

jlegrone commented on stable/jenkins/templates/config.yaml:
```diff
@@ -155,6 +155,20 @@ data:
+    <jenkins.model.JenkinsLocationConfiguration>
+      <adminAddress>{{ default "nobody@nowhere.io" .Values.Master.JenkinsAdminEmail }}</adminAddress>
+{{- if .Values.Master.HostName }}
+{{- if .Values.Master.JenkinsUriSecure }}
```
Where is this coming from? Could we use `Master.Ingress.TLS` instead?

rbutcher replied:
You are totally right. There is no instance I can think of where I would not configure TLS on the ingress and want an https:// uri.

jlegrone commented on stable/jenkins/templates/config.yaml:
```diff
@@ -155,6 +155,20 @@ data:
+  jenkins.model.JenkinsLocationConfiguration.xml: |-
+    <?xml version='1.1' encoding='UTF-8'?>
+    <jenkins.model.JenkinsLocationConfiguration>
+      <adminAddress>{{ default "nobody@nowhere.io" .Values.Master.JenkinsAdminEmail }}</adminAddress>
```
Please document `JenkinsAdminEmail` in the README.  Also, could we just default to an empty string?  Not sure who owns nowhere.io...

jlegrone reviewed (changes requested):
Thanks for this!  It'll be nice to get rid of the warning.  Please see my comments and be sure to bump the chart version as well.

rbutcher committed:
Added JenkinsAdminEmail docs and fixed issues with location config

- Added docs for Master.JenkinsAdminEmail
- Set default for Master.JenkinsAdminEmail to empty string
- Changed conditional to check for Master.Ingress.TLS

Commit:
Update Chart.yaml

Commit:
Merge branch 'master' into set-jenkins-root-url
